src/main/scala/org/coroutines/AstCanonicalization.scala

```diff
@@ -122,6 +121,48 @@ trait AstCanonicalization[C <: Context] {
       (decls, q"$localvarname")
     case q"$selector[..$tpts](...$paramss)" if tpts.length > 0 || paramss.length > 0 =>
       // application
+
+      // We first need to see which parameters, if any, are by-name. Those that are
+      // should not be canonicalized.
+
+      // Maps to the parameter lists, saying whether or not they are by-name.
+      val byNameParams: immutable.Seq[immutable.Seq[Boolean]] = {
+        // Check that `selector` has a non-trivial symbol. If it doesn't, we assume
+        // that there were no by-name parameters.
+        if (selector.symbol != null && selector.symbol != NoSymbol) {
+          val paramLists = selector.symbol.asMethod.paramLists
+
+          // This value does not contain repeated parameters.
```

Comment:
Note that repeated by name parameters are not allowed in Scala.

Also, the `paramss` value might not contain parameters with default values.